x/lockup/types/msgs.go

```diff
@@ -92,3 +93,27 @@ func (m MsgBeginUnlocking) GetSigners() []sdk.AccAddress {
 	owner, _ := sdk.AccAddressFromBech32(m.Owner)
 	return []sdk.AccAddress{owner}
 }
+
+// NewMsgEditLockup creates a message to edit the properties of existing locks
+func NewMsgEditLockup(owner sdk.AccAddress, id uint64, duration time.Duration) *MsgEditLockup {
+	return &MsgEditLockup{
+		Owner:    owner.String(),
+		ID:       id,
+		Duration: duration,
+	}
+}
+
+func (m MsgEditLockup) Route() string { return RouterKey }
+func (m MsgEditLockup) Type() string  { return TypeMsgEditLockup }
+func (m MsgEditLockup) ValidateBasic() error {
+	return nil
```

Comment:
There's zero validation we want to do here? What about checking the fields for basic bounds?

Reply:
Yup seems like adding nil/empty checks will be needed!

Reply:
What about making sure the owner is empty?